bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/Utils.java

```diff
@@ -211,82 +206,30 @@ public static String generateId(String prefix, String path) {
-     * libraries).
-     *
-     * @param resolverFactory the {@link ResourceResolverFactory}
-     *
-     * @return a {@link ResourceResolver} that is able to read information from the components
-     */
-    @Nullable
-    public static ResourceResolver getComponentsResolver(@NotNull ResourceResolverFactory resolverFactory) {
-        try {
-            return resolverFactory.getServiceResourceResolver(
-                    Collections.singletonMap(ResourceResolverFactory.SUBSERVICE, COMPONENTS_SERVICE));
-        } catch (LoginException e) {
-            LOG.error("Cannot login as a service user", e);
-            return null;
-        }
-    }
-
-    /**
-     * Gets the component resource for a given path
-     *
-     * @param path - the path
-     * @param request - the request
-     * @param resolverFactory - the resolver factory
-     *
-     * @return the corresponding resource
-     */
-    @Nullable
-    public static Resource getResource(String path, @NotNull SlingHttpServletRequest request, @NotNull ResourceResolverFactory resolverFactory) {
-        if (path == null) {
-            return null;
-        }
-        ResourceResolver resolver = Utils.getComponentsResolver(resolverFactory);
-        if (resolver == null) {
-            resolver = request.getResourceResolver();
-        }
-        return resolver.getResource(path);
-    }
-
     /**
      * Returns all the super-types of a component defined by its resource type.
      *
-     * @param resourceType the resource type of the component
-     * @param request the current request
-     * @param resolverFactory the {@link ResourceResolverFactory}
+     * @param resourceType the resource type of the component.
+     * @param resourceResolver the resource resolver.
      *
-     * @return a set of the inherited resource types
+     * @return a set of the inherited resource types.
      */
     @NotNull
-    public static Set<String> getSuperTypes(@NotNull String resourceType, @NotNull SlingHttpServletRequest request, ResourceResolverFactory resolverFactory) {
-        Set<String> superTypes = new HashSet<>();
-        addSuperTypes(resourceType, superTypes, request, resolverFactory);
-        return superTypes;
-    }
-
-    private static void addSuperTypes(@NotNull String resourceType, @NotNull Set<String> superTypes, @NotNull SlingHttpServletRequest request, ResourceResolverFactory resolverFactory) {
-        Resource resource = getResource(resourceType, request, resolverFactory);
-        if (resource != null) {
-            String superType = resource.getResourceSuperType();
-            if (StringUtils.isNotEmpty(superType) && !superTypes.contains(superType)) {
-                superTypes.add(superType);
-                addSuperTypes(superType, superTypes, request, resolverFactory);
-            }
-        }
+    public static Set<String> getSuperTypes(@NotNull String resourceType, @NotNull final ResourceResolver resourceResolver) {
```

Comment:
I would prefer keeping the former implementation and not using the Java stream, which I find harder to debug and harder to follow with a recursion.